File: backend/src/modules/treino/entity/Treino.js
class Treino {
  constructor({ id_treino, descricao, data_criacao, instrutor_id_Instrutor, aluno_id_aluno }) {
    this.id_treino = id_treino;
    this.descricao = descricao;
    this.data_criacao = data_criacao;
    this.instrutor_id_Instrutor = instrutor_id_Instrutor;
    this.aluno_id_aluno = aluno_id_aluno;
  }
  validate() {
    const errors = [];
    if (!this.aluno_id_aluno) errors.push('Aluno é obrigatório');
    return errors;
  }
}
module.exports = Treino;

File: backend/src/main/java/com/vitalativa/entity/Treino.java
package com.vitalativa.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;

@Entity
@Table(name = "treino")
public class Treino {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_treino")
    private Integer idTreino;
    
    private String descricao;
    
    @Column(name = "data_criacao")
    private LocalDate dataCriacao;
    
    @Column(name = "instrutor_id_instrutor")
    private Integer instrutorIdInstrutor;
    
    @NotNull(message = "Aluno é obrigatório")
    @Column(name = "aluno_id_aluno")
    private Integer alunoIdAluno;
    
    public Treino() {}
    
    public Integer getIdTreino() { return idTreino; }
    public void setIdTreino(Integer idTreino) { this.idTreino = idTreino; }
    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }
    public LocalDate getDataCriacao() { return dataCriacao; }
    public void setDataCriacao(LocalDate dataCriacao) { this.dataCriacao = dataCriacao; }
    public Integer getInstrutorIdInstrutor() { return instrutorIdInstrutor; }
    public void setInstrutorIdInstrutor(Integer instrutorIdInstrutor) { this.instrutorIdInstrutor = instrutorIdInstrutor; }
    public Integer getAlunoIdAluno() { return alunoIdAluno; }
    public void setAlunoIdAluno(Integer alunoIdAluno) { this.alunoIdAluno = alunoIdAluno; }
}